app/utils/ios-helpers.tsx

```diff
@@ -9,3 +10,75 @@ export function donateRouteIntent(originId: string, destinationId: string) {
 export function reloadAllTimelines() {
   RNBetterRail.reloadAllTimelines()
 }
+
+export function monitorLiveActivities() {
+  RNBetterRail.monitorActivities()
+}
+
+function prepareDataForLiveActivities(route: RouteItem) {
+  // We need to modify the route the fit the original data structure, which we use in the native code
+  const modifiedRoute = { ...route }
+  // @ts-expect-error
+  modifiedRoute.arrivalTime = route.arrivalTimeString
+  // @ts-expect-error
+  modifiedRoute.departureTime = route.departureTimeString
+
+  modifiedRoute.trains = modifiedRoute.trains.map((train) => {
+    const modifiedTrain = { ...train }
+    // @ts-expect-error
+    modifiedTrain.arrivalTime = train.arrivalTimeString
+
+    // @ts-expect-error
+    modifiedTrain.departureTime = train.departureTimeString
+
+    // @ts-expect-error
+    modifiedTrain.destPlatform = train.destinationPlatform
+
+    // @ts-expect-error
+    modifiedTrain.orignStation = train.originStationId
+
+    // @ts-expect-error
+    modifiedTrain.destinationStation = train.destinationStationId
+
+    modifiedTrain.stopStations = train.stopStations.map((stopStation) => {
+      const modifiedStopStation = { ...stopStation }
+      // @ts-expect-error
+      modifiedStopStation.arrivalTime = stopStation.arrivalTimeString
+      // @ts-expect-error
```

Comment:
Why all the `// @ts-expect-error`?

Reply:
> We need to modify the route the fit the original data structure, which we use in the native code

Since we override properties which type doesn't match the value we're putting inside. 
For example, `arrivalTime` is a number, but the original value that comes from the server is a string - and this is also the value which the native code expects us to pass.

I need to refactor the API response modifications in the JS side, but it's a wide project refactor so I preferred to leave it aside for later.